lib/auth.jsx

```diff
@@ -124,4 +188,4 @@ const mapStateToProps = state => ( {
 	hasLoginError: hasLoginError( state ),
 } );
 
-export default connect( mapStateToProps )( Auth );
+export default connect( mapStateToProps, null, null, { pure: false } )( Auth );
```

Comment:
are we sure we need this? using local state is fine. the bigger problem is when we are mutating properties on some object where the object's reference stays the same…

```js
render() {
  this.state.list.push( 'a' );

  return <div>{ this.state.list.join() }</div>
}
```

in this case, a shallow comparison won't find any change but we've appended an item to the list, confusing React/Redux.

it should be fine without `{ pure: false }` I think. it seemed to test okay